File: server/game/ctac/models/Statuses.js
import _ from 'lodash';

import Enum from 'enum';

var Statuses = Enum({
  None       : 0,
  Silence    : 1,
  Shield     : 2,
  Petrify    : 4,
  Taunt      : 8,
  Cloak      : 16,
  Elusive    : 32,
  Root       : 64,
  Charge     : 128,
  CantAttack : 256,
  DyadStrike : 512,
  Flying     : 1024,
  Airdrop    : 2048,
  Cleave     : 4096,
  Piercing   : 8192,
});
export default Statuses;

export function fromInt(value){
  return (_.invert(Statuses))[value];
}

File: centauri-tac/Assets/Scripts/game/models/Statuses.cs
using System;

namespace ctac
{
    [Flags]
    public enum Statuses
    {
        None       = 0,
        Silence    = 1 << 0,
        Shield     = 1 << 1,
        Paralyze   = 1 << 2,
        Taunt      = 1 << 3,
        Cloak      = 1 << 4,
        TechResist = 1 << 5,
        Root       = 1 << 6,
        Charge     = 1 << 7,
        CantAttack = 1 << 8,
        DyadStrike = 1 << 9,
        Flying     = 1 << 10,
        Airdrop    = 1 << 11,
        Cleave     = 1 << 12,
        Piercing   = 1 << 13,


        //The rest of these are fake client side only statuses derived from other properties
        //Used mainly for consistency in showing the status icons
        hasDeathEvent = 1 << 20,
        hasEvent      = 1 << 21,
        isRanged      = 1 << 22,
        hasAura       = 1 << 23

    }
}
